app/queries/servers/channel.ts

```diff
@@ -244,6 +255,39 @@ export const addChannelMembership = async (operator: ServerDataOperator, userId:
     }
 };
 
-export const queryAllMyChannelMembershipIds = async (database: Database, userId: string) => {
-    return database.get(CHANNEL_MEMBERSHIP).query(Q.where('user_id', Q.eq(userId))).fetchIds();
+export const queryUsersOnChannel = (database: Database, channelId: string) => {
+    return database.get<UserModel>(USER).query(Q.on(CHANNEL_MEMBERSHIP, Q.where('channel_id', channelId)));
+};
+
+export const queryChannelsByTypes = (database: Database, channelTypes: ChannelType[]) => {
+    return database.get<ChannelModel>(CHANNEL).query(
+        Q.where('type', Q.oneOf(channelTypes)));
+};
+
+export const queryUserChannelsByTypes = (database: Database, userId: string, channelTypes: ChannelType[]) => {
+    return database.get<ChannelModel>(CHANNEL).query(
+        Q.where('type', Q.oneOf(channelTypes)),
+        Q.on(CHANNEL_MEMBERSHIP, Q.where('user_id', userId)));
+};
+
+export const queryTeamDefaultChannel = (database: Database, teamId: string) => {
+    return database.get<ChannelModel>(MM_TABLES.SERVER.CHANNEL).query(
+        Q.where('team_id', teamId),
+        Q.where('name', General.DEFAULT_CHANNEL),
+    );
+};
+
+export const queryMyChannelsByTeam = (database: Database, teamId: string) => {
+    return database.get<MyChannelModel>(MY_CHANNEL).query(
+        Q.on(CHANNEL, Q.and(
+            Q.where('delete_at', Q.eq(0)),
```

Comment:
I wonder if you should return everything and then deal with filtering the archived channels base on need or better yet make it optional.